pkgs/applications/emulators/pcsx-redux/default.nix

```diff
@@ -0,0 +1,76 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, zlib
+, pkg-config
+, libuv
+, curl
+, ffmpeg
+, glfw
+, capstone
+, freetype
+, libX11
+, imagemagick
+, libpulseaudio
+, alsa-lib
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pcsx-redux";
+  version = "unstable-2023-04-18";
+
+  src = fetchFromGitHub {
+    owner = "grumpycoders";
+    repo = "pcsx-redux";
+    fetchSubmodules = true;
+    rev = "0093bea292920f1d1c3eb7515026b4153fe2e926";
+    hash = "sha256-Mwg4jAve0nLuODV/8JO4t5akswi3/qRU/Q5UihJnbps=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [
+    imagemagick
+    libX11
+    freetype
+    capstone
+    glfw
+    ffmpeg
+    curl
+    libuv
+    zlib
+  ];
+
+  # A dependency in third_party (miniaudio) tries to load audio backends on
+  # demand, by querying dynamic libraries at runtime. Patch the paths to point
+  # at ALSA and PulseAudio backends so that it has a chance of finding them, say
+  # on NixOS. There are more backends, such as OSS and JACK. I guess we can add
+  # these on demand in the future.
+  postPatch = lib.optionalString stdenv.isLinux ''
+    substituteInPlace third_party/miniaudio/miniaudio.h \
+      --replace 'libasound.so' '${alsa-lib}/lib/libasound.so'
+    substituteInPlace third_party/miniaudio/extras/miniaudio_split/miniaudio.c \
+      --replace 'libasound.so' '${alsa-lib}/lib/libasound.so'
+    substituteInPlace third_party/miniaudio/miniaudio.h \
+      --replace 'libpulse.so' '${libpulseaudio}/lib/libpulse.so'
+    substituteInPlace third_party/miniaudio/extras/miniaudio_split/miniaudio.c \
+      --replace 'libpulse.so' '${libpulseaudio}/lib/libpulse.so'
+  '';
+
+  # Fails with -Wformat-security: https://github.com/grumpycoders/pcsx-redux/issues/1250
```

Comment:
all the format-security errors got fixed (https://github.com/grumpycoders/pcsx-redux/pull/1309 and https://github.com/grumpycoders/pcsx-redux/pull/1632) and https://github.com/grumpycoders/pcsx-redux/pull/1635 should prevent new ones in the future

Reply:
Cool. I'm probably not going to update this PR very soon so feel free to take anything you need from it and open up a new version if needed, then I'll close this one.